pkg/server/job_profiler_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package server
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/base"
+	_ "github.com/cockroachdb/cockroach/pkg/cloud/impl" // register ExternalStorage providers.
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/server/serverpb"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/stretchr/testify/require"
+)
+
+// fakeExecResumer calls optional callbacks during the job lifecycle.
+type fakeExecResumer struct {
+	OnResume     func(context.Context) error
+	FailOrCancel func(context.Context) error
+}
+
+func (d fakeExecResumer) ForceRealSpan() bool {
+	return true
+}
+
+func (d fakeExecResumer) DumpTraceAfterRun() bool {
+	return true
+}
+
+var _ jobs.Resumer = fakeExecResumer{}
+var _ jobs.TraceableJob = fakeExecResumer{}
+
+func (d fakeExecResumer) Resume(ctx context.Context, execCtx interface{}) error {
+	if d.OnResume != nil {
+		if err := d.OnResume(ctx); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (d fakeExecResumer) OnFailOrCancel(ctx context.Context, _ interface{}, _ error) error {
+	if d.FailOrCancel != nil {
+		return d.FailOrCancel(ctx)
+	}
+	return nil
+}
+
+// TestJobExecutionDetailsRouting tests that the request job execution details
+// endpoint redirects the request to the current coordinator node of the job.
+func TestJobExecutionDetailsRouting(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	hasStartedCh := make(chan struct{})
+	defer close(hasStartedCh)
+	jobs.RegisterConstructor(jobspb.TypeImport, func(j *jobs.Job, _ *cluster.Settings) jobs.Resumer {
+		return fakeExecResumer{
+			OnResume: func(ctx context.Context) error {
+				hasStartedCh <- struct{}{}
+				return nil
+			},
+		}
+	}, jobs.UsesTenantCostControl)
+	defer jobs.ResetConstructors()()
+
+	dialedNodeID := roachpb.NodeID(-1)
+	ctx := context.Background()
+	tc := serverutils.StartCluster(t, 2, base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+			Knobs: base.TestingKnobs{
+				JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
+				Server: &TestingKnobs{
+					DialNodeCallback: func(ctx context.Context, nodeID roachpb.NodeID) error {
```

Comment:
if there is an easier way to check that a particular node was dialed and returned us the response i'm all ears, i looked around but couldn't find any